backend/executor.go

```diff
@@ -310,14 +311,58 @@ func (g *grpcExecutor) StartInstance(ctx context.Context, req *protos.CreateInst
 	ctx, span := helpers.StartNewCreateOrchestrationSpan(ctx, req.Name, req.Version.GetValue(), instanceID)
 	defer span.End()
 
+	// retreive instance with instanceID
+	metadata, err := g.backend.GetOrchestrationMetadata(ctx, api.InstanceID(instanceID))
```

Comment:
If possible, I'd like to find a way to accomplish our goal without checking the status first. I regret how we did this in Durable Functions because it leads to non-deterministic behavior since there are race conditions with status checks. It would be better if we did all this checking as part of the internal implementation of the backend function for orchestration creation.

Reply:
@cgillum please correct me if I don't get it correct. The idea is that we should limit the number of times that we query the database as there could be data discrepancies between queries we made, so the best way is to query/update the database in one transaction so we always get consistent data for the next step. Thanks.

Reply:
> It would be better if we did all this checking as part of the internal implementation of the backend function for orchestration creation.
I was trying to avoid this as this requires updates from the backend implementation side. I was trying to code the logic from the caller side, so it asks for minimal updates from different implementations of the backend.

Reply:
I'm not too worried about limiting the number of times we query the database. However, in this case, the status check goes over gRPC and will be traced, making traces a bit noisier. But actually my main concern is about race conditions. For example, if two clients call `StartInstance` at the same time, we won't be able to correctly detect this. However, if we do the implementation directly in the backend, we can leverage database transactions to ensure that we always make decisions with the most up-to-date information, avoiding any race conditions.